src/components/FileUpload/FileUpload.tsx

```diff
@@ -0,0 +1,475 @@
+          background-color: ${({ theme }) =>
+            theme.click.card.secondary.color.background.hover};
+        }
+      }
+
+      ${props.$isDragging &&
+      css`
+        background-color: ${({ theme }) =>
+          theme.click.card.secondary.color.background.hover};
+        border-color: #9ca3af;
+      `}
+    `}
+
+  ${props =>
+    props.$isError &&
+    css`
+      background-color: ${({ theme }) => theme.click.fileUpload.color.background.error};
+      border: none;
+    `}
+`;
+
+const FileUploadTitle = styled(Title)`
+  font-family: ${({ theme }) => theme.click.fileUpload.typography.title.default};
+  color: ${({ theme }) => theme.click.fileUpload.color.title.default};
+`;
+
+const FileUploadDescription = styled(Text)`
+  font-family: ${({ theme }) => theme.click.fileUpload.typography.description.default};
+  color: ${({ theme }) => theme.click.fileUpload.color.description.default};
+`;
+
+const DocumentIcon = styled(Icon)`
+  svg {
+    width: ${({ theme }) => theme.click.fileUpload.sm.icon.size.width};
+    height: ${({ theme }) => theme.click.fileUpload.sm.icon.size.height};
+    color: ${({ theme }) => theme.click.fileUpload.sm.color.icon.default};
+  }
+`;
+
+const UploadIcon = styled(Icon)`
+  svg {
+    width: ${({ theme }) => theme.click.fileUpload.md.icon.size.width};
+    height: ${({ theme }) => theme.click.fileUpload.md.icon.size.height};
+    color: ${({ theme }) => theme.click.fileUpload.md.color.icon.default};
+  }
+`;
+
+const UploadText = styled.div<{ $size: "sm" | "md"; $hasFile: boolean }>`
+  text-align: ${props => (props.$hasFile || props.$size === "sm" ? "left" : "center")};
+  ${props =>
+    (props.$hasFile || props.$size === "sm") &&
+    css`
+      flex: 1;
+      margin-left: 12px;
+    `}
+
+  ${props =>
+    !props.$hasFile &&
+    props.$size === "md" &&
+    css`
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+      width: 100%;
+    `}
+`;
+
+const FileInfoHeader = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 12px;
+  width: 100%;
+`;
+
+const FileInfo = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 4px;
+  flex: 1;
+`;
+
+const FileDetails = styled.div`
+  display: flex;
+  gap: ${({ theme }) => theme.click.fileUpload.sm.space.gap};
+  border: none;
+`;
+
+const FileActions = styled.div`
+  display: flex;
+  align-items: center;
+  margin-left: auto;
+  gap: 0;
+`;
+
+const ProgressContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  gap: 8px;
+  margin-top: 4px;
+  width: 100%;
+  padding-left: 4px;
+  padding-right: 8px;
+`;
+
+const ProgressBarContainer = styled.div`
+  width: 100%;
+  height: 4px;
+  background-color: #e5e7eb;
+  border-radius: 2px;
+  flex: 1;
+  overflow: hidden;
+`;
+
+const ProgressBar = styled.div<{ $progress: number }>`
+  height: 100%;
+  width: ${props => props.$progress}%;
+  transition: width 0.3s ease;
+
+  ${({ theme }) =>
+    theme.name === "dark"
```

Comment:
I'm updating the tokens, once we have them, there is no need for light/dark theme to be set here.